presto-main/src/main/java/io/prestosql/security/NonModifyingStatementAccessControl.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.security;
+
+import io.prestosql.spi.StatementRewriteContext;
+import io.prestosql.spi.security.AccessDeniedException;
+import io.prestosql.spi.security.StatementAccessControl;
+import io.prestosql.spi.security.StatementAccessControlFactory;
+
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class NonModifyingStatementAccessControl
```

Comment:
remove this class, not needed

Reply:
If StatementAccessControlManager can be passed, then this can be removed. Let me check and update

Reply:
> If StatementAccessControlManager can be passed

Even if it's not passed you could just use `Optional` in `QueryPreparer`.

Reply:
Removed NonModifyingStatementAccessControl.